tools/lint.py

```
@@ -2021,74 +2044,75 @@ def check_delivery(root: Path) -> list[str]:
+    if not manifest.is_file():
+        return [
+            "marketplace-source: .claude-plugin/marketplace.json is missing -- "
+            "Codex and Claude must discover the shared tradecraft plugin from "
+            "one marketplace manifest"
+        ]
+    try:
+        content = manifest.read_text(encoding="utf-8", errors="replace")
+    except OSError as exc:
+        return [
+            "marketplace-source: .claude-plugin/marketplace.json cannot be read "
+            f"({exc}) -- the shared tradecraft source cannot be verified"
```

Ruff (0.16.2)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)